.github/workflows/coverage.yml

```diff
@@ -0,0 +1,32 @@
+on:
+  push:
+    branches: [master]
+  pull_request:
+
+name: Code Coverage
+
+jobs:
+  coverage:
+    name: Test Coverage
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: checkout source
+        uses: actions/checkout@v2
+
+      - name: Install nightly toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+          override: true
+
+      - name: Run cargo-tarpaulin
+        uses: actions-rs/tarpaulin@v0.1
+        with:
+          version: '0.13.2'
+
+      - name: Upload to codecov.io
+        uses: codecov/codecov-action@v1.0.7
+        with:
+          token: ${{secrets.CODECOV_TOKEN}}
```

Comment:
Even less thrilled about sending my code to random places I(or github?) need tokens for. I know it's all public, but I'd rather keep it local if possible.

Reply:
so this doesn't send code anywhere, just a coverage report. The token is only used for publishing coverage reports to codecov. If this token leaked, the most damaging thing anyone could do to you is spoof the coverage reports for this particular repo (until you revoked the token). exceptionally low risk.
the benefit is that codecov will comment on pull requests to provide a graphical coverage report. kind of cool.

looks like this - https://github.com/danieleades/longshoreman/pull/26#issuecomment-638445279